### ISSUE TITLE
Error visualization

### ISSUE THREAD

**Issue by mhuebert** [mhuebert/maria#73]
Just tabling a number of cross-cutting concerns to discuss in-person sometime.

- Prioritize messages & collapse most by default: we usually have a number of messages at once, coming from warnings & errors, which need to be prioritized and visualized appropriately. We don't necessarily want to show all of the errors at once - probably only show the top message, with a link to click for more. I started today by putting 'warnings' above 'errors' because they are often more informative about root-causes (issues that have been noticed before compile is finished).
- Names for warnings/errors: I modified `parse-warning` and it now returns the :name and :doc(string) for a warning. I find that I don't want to have to read a whole paragraph for warnings/errors that I see often - I would rather scan & see the name of the error first, and then read the details if necessary. This would work well for the collapsed view - we could show a summary of all the errors/warnings by name, & click to see more.
- Links in warnings/errors: it would be nice to be able to link to further documentation to help a user with their error message. Unsure if it would be more convenient to just parse error messages as markdown (with links) or store links as some kind of structured metadata (easier to format nicely).
- Stack visualization and expansion: we should parse the error message stack and allow the user to view source code (with location highlighted) for each item in the stack. Stack should not be expanded/shown by default.
- Design: I have made some add-hoc modifications (dividers, name display) to the error/warning views but they could use a more holistic revisit as it is not looking very pretty. (This would be a natural part of implementing summary/expandable views.)
- Code highlights: it should be possible for a message-handler to return instructions for which form(s) to highlight, given an error/warning + context.

So, in summary:

- [ ] Show only top message, click to view the rest
- [ ] Return a :name for each error for summary view
- [ ] Support linking to external resources from error messages
- [ ] Nicer overall design for errors/warnings
- [ ] Friendly display/navigation of the error stack
- [ ] Support specific form highlighting based on the particular error

**Comment by mhuebert** [comment]
Moved to [Roadmap and wishlist](https://github.com/mhuebert/maria/wiki/Roadmap-and-wishlist)
